diff --git a/lib/curlx/multibyte.c b/lib/curlx/multibyte.c
--- a/lib/curlx/multibyte.c
+++ b/lib/curlx/multibyte.c
@@ -42,2 +42,18 @@
 
+UINT curlx_utf8_codepage()
+{
+#if _MSC_VER
+#pragma warning(push)
+#pragma warning(disable : 4996)
+#endif
+    BYTE win_major = LOBYTE(LOWORD(GetVersion()));
+    /* BYTE win_minor = HIBYTE(LOWORD(GetVersion())); */
+#if _MSC_VER
+#pragma warning(pop)
+#endif
+
+    /* NT4 doesn't support UTF-8 code page */
+    return win_major >= 5 ? CP_UTF8 : CP_ACP;
+}
+
 wchar_t *curlx_convert_UTF8_to_wchar(const char *str_utf8)
@@ -47,3 +63,3 @@
   if(str_utf8) {
-    int str_w_len = MultiByteToWideChar(CP_UTF8, MB_ERR_INVALID_CHARS,
+    int str_w_len = MultiByteToWideChar(curlx_utf8_codepage(), MB_ERR_INVALID_CHARS,
                                         str_utf8, -1, NULL, 0);
@@ -52,3 +68,3 @@
       if(str_w) {
-        if(MultiByteToWideChar(CP_UTF8, 0, str_utf8, -1, str_w,
+        if(MultiByteToWideChar(curlx_utf8_codepage(), 0, str_utf8, -1, str_w,
                                str_w_len) == 0) {
@@ -69,3 +85,3 @@
   if(str_w) {
-    int bytes = WideCharToMultiByte(CP_UTF8, 0, str_w, -1,
+    int bytes = WideCharToMultiByte(curlx_utf8_codepage(), 0, str_w, -1,
                                     NULL, 0, NULL, NULL);
@@ -74,3 +90,3 @@
       if(str_utf8) {
-        if(WideCharToMultiByte(CP_UTF8, 0, str_w, -1, str_utf8, bytes,
+        if(WideCharToMultiByte(curlx_utf8_codepage(), 0, str_w, -1, str_utf8, bytes,
                                NULL, NULL) == 0) {

diff --git a/src/tool_cb_wrt.c b/src/tool_cb_wrt.c
--- a/src/tool_cb_wrt.c
+++ b/src/tool_cb_wrt.c
@@ -43,2 +43,19 @@
 
+UINT tool_utf8_codepage()
+{
+#if _MSC_VER
+#pragma warning(push)
+#pragma warning(disable : 4996)
+#endif
+    BYTE win_major = LOBYTE(LOWORD(GetVersion()));
+    /* BYTE win_minor = HIBYTE(LOWORD(GetVersion())); */
+#if _MSC_VER
+#pragma warning(pop)
+#endif
+
+    /* NT4 doesn't support UTF-8 code page */
+    return win_major >= 5 ? CP_UTF8 : CP_ACP;
+}
+
+
 /* create/open a local file for writing, return TRUE on success */
@@ -165,3 +182,3 @@
 
-      if(MultiByteToWideChar(CP_UTF8, 0, (LPCSTR)outs->utf8seq, -1,
+      if(MultiByteToWideChar(tool_utf8_codepage(), 0, (LPCSTR)outs->utf8seq, -1,
                              prefix, CURL_ARRAYSIZE(prefix))) {
@@ -207,3 +224,3 @@
     /* calculate buffer size for wide characters */
-    DWORD len = (DWORD)MultiByteToWideChar(CP_UTF8, 0, (LPCSTR)rbuf,
+    DWORD len = (DWORD)MultiByteToWideChar(tool_utf8_codepage(), 0, (LPCSTR)rbuf,
                                            (int)rlen, NULL, 0);
@@ -222,3 +239,3 @@
 
-    len = (DWORD)MultiByteToWideChar(CP_UTF8, 0, (LPCSTR)rbuf, (int)rlen,
+    len = (DWORD)MultiByteToWideChar(tool_utf8_codepage(), 0, (LPCSTR)rbuf, (int)rlen,
                                      global->term.buf,
