diff -git a/src/tool_operate.c b/src/tool_operate.c
--- a/src/tool_operate.c
+++ b/src/tool_operate.c
@@ -586,6 +586,12 @@
         /* ftruncate is not available, so just reposition the file
            to the location we would have truncated it. */
         rc = curlx_fseek(outs->stream, outs->init, SEEK_SET);
+#if defined(_WIN32)
+        /* nscurl: mark the end-of-file */
+        if(!SetEndOfFile((HANDLE)_get_osfhandle(outs->stream))) {
+          rc = -1;
+        }
+#endif
 #endif
         if(rc) {
           errorf("Failed seeking to end of file");
