diff --git a/include/curl/curl.h b/include/curl/curl.h
--- a/include/curl/curl.h
+++ b/include/curl/curl.h
@@ -81,3 +81,7 @@
 #include <winsock2.h>
+#if _WIN32_WINNT >= 0x0501
 #include <ws2tcpip.h>
+#else
+#include <wspiapi.h>
+#endif
 #endif
 
diff --git a/lib/setup-win32.h b/lib/setup-win32.h
--- a/lib/setup-win32.h
+++ b/lib/setup-win32.h
@@ -91,3 +91,7 @@
 #  include <winsock2.h>
-#  include <ws2tcpip.h>
+#  if _WIN32_WINNT >= 0x0501
+#    include <ws2tcpip.h>
+#  else
+#    include <wspiapi.h>
+#  endif
 #  include <windows.h>
