diff --git a/lib/curlx/fopen.c b/lib/curlx/fopen.c
--- a/lib/curlx/fopen.c
+++ b/lib/curlx/fopen.c
@@ -63,12 +63,14 @@
   wchar_t *str_w = NULL;
 
   if(str_utf8) {
-    int str_w_len = MultiByteToWideChar(CP_UTF8, MB_ERR_INVALID_CHARS,
+    /* [nscurl] CP_UTF8 not available in NT4 */
+    int str_w_len = MultiByteToWideChar(nscurl_utf8_codepage(), MB_ERR_INVALID_CHARS,
                                         str_utf8, -1, NULL, 0);
     if(str_w_len > 0) {
       str_w = CURLX_MALLOC(str_w_len * sizeof(wchar_t));
       if(str_w) {
-        if(MultiByteToWideChar(CP_UTF8, 0,
+        /* [nscurl] CP_UTF8 not available in NT4 */
+        if(MultiByteToWideChar(nscurl_utf8_codepage(), 0,
                                str_utf8, -1, str_w, str_w_len) == 0) {
           CURLX_FREE(str_w);
           return NULL;
