diff --git a/lib/rand.c b/lib/rand.c
--- a/lib/rand.c
+++ b/lib/rand.c
@@ -34,6 +34,7 @@
 #include "escape.h"
 
 #ifdef _WIN32
+#if defined(_WIN32_WINNT) && (_WIN32_WINNT >= _WIN32_WINNT_VISTA)   /* [nscurl] prevent using bcrypt in NT4 builds */
 #include <bcrypt.h>
 #ifndef STATUS_SUCCESS
 #define STATUS_SUCCESS ((NTSTATUS)0x00000000L)
@@ -49,6 +50,7 @@
 
   return CURLE_OK;
 }
+#endif  /* nscurl */
 #endif
 
 #ifndef USE_SSL
@@ -62,12 +64,14 @@
 
   /* Trying cryptographically secure functions first */
 #ifdef _WIN32
+#if defined(_WIN32_WINNT) && (_WIN32_WINNT >= _WIN32_WINNT_VISTA)   /* [nscurl] prevent using bcrypt in NT4 builds */
   (void)data;
   {
     CURLcode result = Curl_win32_random(entropy, length);
     if(result != CURLE_NOT_BUILT_IN)
       return result;
   }
+#endif  /* nscurl */
 #endif
 
 #ifdef HAVE_ARC4RANDOM
