diff --git a/src/tool_doswin.c b/src/tool_doswin.c
index db2b8b78a..864aa1418 100644
--- a/src/tool_doswin.c
+++ b/src/tool_doswin.c
@@ -615,2 +615,3 @@
 #if !defined(CURL_WINDOWS_UWP) && !defined(UNDER_CE)
+#if _WIN32_WINNT >= _WIN32_WINNT_WIN2K      /* [nscurl] Toolhelp functions unavailable in NT4 */
   HANDLE hnd = INVALID_HANDLE_VALUE;
@@ -659,2 +660,3 @@
     CloseHandle(hnd);
+#endif
 #endif
