diff --git a/src/tool_doswin.c b/src/tool_doswin.c
--- a/src/tool_doswin.c
+++ b/src/tool_doswin.c
@@ -595,6 +595,7 @@
 {
   struct curl_slist *slist = NULL;
 #ifndef CURL_WINDOWS_UWP
+#if _WIN32_WINNT >= _WIN32_WINNT_WIN2K      /* [nscurl] Toolhelp functions unavailable in NT4 */
   HANDLE hnd = INVALID_HANDLE_VALUE;
   MODULEENTRY32 mod = { 0 };
 
@@ -639,6 +640,7 @@
 cleanup:
   if(hnd != INVALID_HANDLE_VALUE)
     CloseHandle(hnd);
+#endif
 #endif
   return slist;
 }
