diff --git a/lib/curlx/multibyte.c b/lib/curlx/multibyte.c
--- a/lib/curlx/multibyte.c
+++ b/lib/curlx/multibyte.c
@@ -36,12 +36,13 @@
   wchar_t *str_w = NULL;
 
   if(str_utf8) {
-    int str_w_len = MultiByteToWideChar(CP_UTF8, MB_ERR_INVALID_CHARS,
+    /* [nscurl] CP_UTF8 not available in NT4 */
+    int str_w_len = MultiByteToWideChar(nscurl_utf8_codepage(), MB_ERR_INVALID_CHARS,
                                         str_utf8, -1, NULL, 0);
     if(str_w_len > 0) {
       str_w = curlx_malloc(str_w_len * sizeof(wchar_t));
       if(str_w) {
-        if(MultiByteToWideChar(CP_UTF8, 0, str_utf8, -1, str_w,
+        if(MultiByteToWideChar(nscurl_utf8_codepage(), 0, str_utf8, -1, str_w,
                                str_w_len) == 0) {
           curlx_free(str_w);
           return NULL;
@@ -58,12 +59,13 @@
   char *str_utf8 = NULL;
 
   if(str_w) {
-    int bytes = WideCharToMultiByte(CP_UTF8, 0, str_w, -1,
+    /* [nscurl] CP_UTF8 not available in NT4 */
+    int bytes = WideCharToMultiByte(nscurl_utf8_codepage(), 0, str_w, -1,
                                     NULL, 0, NULL, NULL);
     if(bytes > 0) {
       str_utf8 = curlx_malloc(bytes);
       if(str_utf8) {
-        if(WideCharToMultiByte(CP_UTF8, 0, str_w, -1, str_utf8, bytes,
+        if(WideCharToMultiByte(nscurl_utf8_codepage(), 0, str_w, -1, str_utf8, bytes,
                                NULL, NULL) == 0) {
           curlx_free(str_utf8);
           return NULL;
