diff --git a/include/internal/common.h b/include/internal/common.h
--- a/include/internal/common.h
+++ b/include/internal/common.h
@@ -235,2 +235,20 @@
 
+# if defined(_WIN32)
+static ossl_inline UINT ossl_utf8_codepage()
+{
+#if _MSC_VER
+#pragma warning(push)
+#pragma warning(disable : 4996)
+#endif
+    BYTE win_major = LOBYTE(LOWORD(GetVersion()));
+    /* BYTE win_minor = HIBYTE(LOWORD(GetVersion())); */
+#if _MSC_VER
+#pragma warning(pop)
+#endif
+
+    /* NT4 doesn't support UTF-8 code page */
+    return win_major >= 5 ? CP_UTF8 : CP_ACP;
+}
+# endif
+
 #endif

diff --git a/crypto/defaults.c b/crypto/defaults.c
--- a/crypto/defaults.c
+++ b/crypto/defaults.c
@@ -102,3 +102,3 @@
 
-    if (!WideCharToMultiByte(CP_UTF8, 0, tempstr, -1, dst, dstsizebytes,
+    if (!WideCharToMultiByte(ossl_utf8_codepage(), 0, tempstr, -1, dst, dstsizebytes,
                              NULL, NULL))

diff --git a/crypto/getenv.c b/crypto/getenv.c
--- a/crypto/getenv.c
+++ b/crypto/getenv.c
@@ -63,3 +63,3 @@
                 /* determine value string size in utf-8 */
-                vallen = WideCharToMultiByte(CP_UTF8, 0, valw, -1, NULL, 0,
+                vallen = WideCharToMultiByte(ossl_utf8_codepage(), 0, valw, -1, NULL, 0,
                                              NULL, NULL);
@@ -73,3 +73,3 @@
             /* convert value string from wide to utf-8 */
-            if (WideCharToMultiByte(CP_UTF8, 0, valw, -1, val, vallen,
+            if (WideCharToMultiByte(ossl_utf8_codepage(), 0, valw, -1, val, vallen,
                                     NULL, NULL) == 0) {

diff --git a/crypto/LPdir_win.c b/crypto/LPdir_win.c
--- a/crypto/LPdir_win.c
+++ b/crypto/LPdir_win.c
@@ -44,2 +44,3 @@
 #endif
+#include "internal/cryptlib.h"
 
@@ -106,3 +107,3 @@
 # ifdef CP_UTF8
-                if ((sz = MultiByteToWideChar((cp = CP_UTF8), 0,
+                if ((sz = MultiByteToWideChar((cp = ossl_utf8_codepage()), 0,
                                               directory, len_0,

diff --git a/crypto/o_fopen.c b/crypto/o_fopen.c
--- a/crypto/o_fopen.c
+++ b/crypto/o_fopen.c
@@ -62,6 +62,6 @@
      */
-    if ((sz = MultiByteToWideChar(CP_UTF8, (flags = MB_ERR_INVALID_CHARS),
+    if ((sz = MultiByteToWideChar(ossl_utf8_codepage(), (flags = MB_ERR_INVALID_CHARS),
                                   filename, len_0, NULL, 0)) > 0 ||
         (GetLastError() == ERROR_INVALID_FLAGS &&
-         (sz = MultiByteToWideChar(CP_UTF8, (flags = 0),
+         (sz = MultiByteToWideChar(ossl_utf8_codepage(), (flags = 0),
                                    filename, len_0, NULL, 0)) > 0)
@@ -71,5 +71,5 @@
 
-        if (MultiByteToWideChar(CP_UTF8, flags,
+        if (MultiByteToWideChar(ossl_utf8_codepage(), flags,
                                 filename, len_0, wfilename, sz) &&
-            MultiByteToWideChar(CP_UTF8, 0, mode, strlen(mode) + 1,
+            MultiByteToWideChar(ossl_utf8_codepage(), 0, mode, strlen(mode) + 1,
                                 wmode, OSSL_NELEM(wmode)) &&

diff --git a/crypto/rand/randfile.c b/crypto/rand/randfile.c
--- a/crypto/rand/randfile.c
+++ b/crypto/rand/randfile.c
@@ -291,6 +291,6 @@
         if (GetEnvironmentVariableW(var, val, envlen) < envlen
-                && (sz = WideCharToMultiByte(CP_UTF8, 0, val, -1, NULL, 0,
+                && (sz = WideCharToMultiByte(ossl_utf8_codepage(), 0, val, -1, NULL, 0,
                                              NULL, NULL)) != 0) {
             s = _alloca(sz);
-            if (WideCharToMultiByte(CP_UTF8, 0, val, -1, s, sz,
+            if (WideCharToMultiByte(ossl_utf8_codepage(), 0, val, -1, s, sz,
                                     NULL, NULL) == 0)

diff --git a/crypto/ui/ui_openssl.c b/crypto/ui/ui_openssl.c
--- a/crypto/ui/ui_openssl.c
+++ b/crypto/ui/ui_openssl.c
@@ -313,3 +313,3 @@
                 wresult[numread] = '\0';
-                if (WideCharToMultiByte(CP_UTF8, 0, wresult, -1,
+                if (WideCharToMultiByte(ossl_utf8_codepage(), 0, wresult, -1,
                                         result, sizeof(result), NULL, 0) > 0)
