diff --git a/Configurations/unix-Makefile.tmpl b/Configurations/unix-Makefile.tmpl
--- a/Configurations/unix-Makefile.tmpl
+++ b/Configurations/unix-Makefile.tmpl
@@ -754,2 +754,18 @@
 	done
+	@$(PERL) $(SRCDIR)/util/mkdir-p.pl "$(DESTDIR)$(INSTALLTOP)/include/openssl/crypto"
+	@set -e; for i in $(SRCDIR)/include/crypto/*.h \
+			  $(BLDDIR)/include/crypto/*.h; do \
+		fn=`basename $$i`; \
+		$(ECHO) "install $$i -> $(DESTDIR)$(INSTALLTOP)/include/openssl/crypto/$$fn"; \
+		cp $$i "$(DESTDIR)$(INSTALLTOP)/include/openssl/crypto/$$fn"; \
+		chmod 644 "$(DESTDIR)$(INSTALLTOP)/include/openssl/crypto/$$fn"; \
+	done
+	@$(PERL) $(SRCDIR)/util/mkdir-p.pl "$(DESTDIR)$(INSTALLTOP)/include/openssl/internal"
+	@set -e; for i in $(SRCDIR)/include/internal/*.h \
+			  $(BLDDIR)/include/internal/*.h; do \
+		fn=`basename $$i`; \
+		$(ECHO) "install $$i -> $(DESTDIR)$(INSTALLTOP)/include/openssl/internal/$$fn"; \
+		cp $$i "$(DESTDIR)$(INSTALLTOP)/include/openssl/internal/$$fn"; \
+		chmod 644 "$(DESTDIR)$(INSTALLTOP)/include/openssl/internal/$$fn"; \
+	done
 	@$(PERL) $(SRCDIR)/util/mkdir-p.pl "$(DESTDIR)$(libdir)"
diff --git a/Configurations/windows-makefile.tmpl b/Configurations/windows-makefile.tmpl
--- a/Configurations/windows-makefile.tmpl
+++ b/Configurations/windows-makefile.tmpl
@@ -564,2 +564,14 @@
 				       "$(INSTALLTOP)\include\openssl"
+	@"$(PERL)" "$(SRCDIR)\util\mkdir-p.pl" "$(INSTALLTOP)\include\openssl\crypto"
+	@"$(PERL)" "$(SRCDIR)\util\copy.pl" "-exclude_re=/__DECC_" \
+				       "$(SRCDIR)\include\crypto\*.h" \
+				       "$(INSTALLTOP)\include\openssl\crypto"
+	@"$(PERL)" "$(SRCDIR)\util\copy.pl" "$(BLDDIR)\include\crypto\*.h" \
+				       "$(INSTALLTOP)\include\openssl\crypto"
+	@"$(PERL)" "$(SRCDIR)\util\mkdir-p.pl" "$(INSTALLTOP)\include\openssl\internal"
+	@"$(PERL)" "$(SRCDIR)\util\copy.pl" "-exclude_re=/__DECC_" \
+				       "$(SRCDIR)\include\internal\*.h" \
+				       "$(INSTALLTOP)\include\openssl\internal"
+	@"$(PERL)" "$(SRCDIR)\util\copy.pl" "$(BLDDIR)\include\internal\*.h" \
+				       "$(INSTALLTOP)\include\openssl\internal"
 	@"$(PERL)" "$(SRCDIR)\util\mkdir-p.pl" "$(libdir)"
