diff --git a/crypto/LPdir_win.c b/crypto/LPdir_win.c
--- a/crypto/LPdir_win.c
+++ b/crypto/LPdir_win.c
@@ -40,2 +40,3 @@
 #include <tchar.h>
+#include "internal/common.h"
 #include "internal/numbers.h"
@@ -62,3 +63,3 @@
 #ifdef CP_UTF8
-#define CP_DEFAULT CP_UTF8
+#define CP_DEFAULT ossl_utf8_codepage()
 #else
@@ -106,3 +107,3 @@
 #ifdef CP_UTF8
-                if ((sz = MultiByteToWideChar((cp = CP_UTF8), 0,
+                if ((sz = MultiByteToWideChar((cp = ossl_utf8_codepage()), 0,
                          directory, (int)len_0,
