diff --git a/providers/implementations/rands/seeding/rand_win.c b/providers/implementations/rands/seeding/rand_win.c
--- a/providers/implementations/rands/seeding/rand_win.c
+++ b/providers/implementations/rands/seeding/rand_win.c
@@ -88,4 +88,9 @@
         /* poll the CryptoAPI PRNG */
+        /* note: CRYPT_SILENT available starting in Windows 2000 */
         if (CryptAcquireContextW(&hProvider, NULL, NULL, PROV_RSA_FULL,
+#if _WIN32_WINNT >= 0x0500
                 CRYPT_VERIFYCONTEXT | CRYPT_SILENT)
+#else
+                CRYPT_VERIFYCONTEXT | (LOBYTE(LOWORD(GetVersion())) >= 5 ? CRYPT_SILENT : 0))
+#endif
             != 0) {
@@ -108,5 +113,10 @@
         /* poll the Pentium PRG with CryptoAPI */
+        /* note: CRYPT_SILENT available starting in Windows 2000 */
         if (CryptAcquireContextW(&hProvider, NULL,
                 INTEL_DEF_PROV, PROV_INTEL_SEC,
+#if _WIN32_WINNT >= 0x0500
                 CRYPT_VERIFYCONTEXT | CRYPT_SILENT)
+#else
+                CRYPT_VERIFYCONTEXT | (LOBYTE(LOWORD(GetVersion())) >= 5 ? CRYPT_SILENT : 0))
+#endif
             != 0) {
